src/compass/s1_cslc_qa.py

```diff
@@ -279,32 +280,136 @@ def valid_pixel_percentages(self, cslc_h5py_root):
         '''
-        rfi_qa_items = [
-            Meta('is_rfi_info_available',True,
-                 'Whether or not RFI information is available'),
-            # Follow key/values only assigned if RFI info is avaiable
-            Meta('rfi_mitigation_performed', True,
-                 'Whether or not the RFI mitigation step was performed'),
-            Meta('rfi_mitigation_domain', '',
-                 'Domain the RFI mitigation step was performed'),
-            Meta('rfi_burst_report', '', 'Burst RFI report')
-        ]
-
-        # create HDF5 group for RFI info
-        h5_rfi_path = f'{QA_PATH}/rfi_information'
-        rfi_group = cslc_h5py_root.require_group(h5_rfi_path)
 
-        # write items to HDF5 and dict
-        _qa_items_to_h5_and_dict(rfi_group, self.rfi_dict, rfi_qa_items)
+        for burst in bursts:
+            is_rfi_info_available = burst.burst_rfi_info is not None
+            rfi_qa_items_pol = [Meta('is_rfi_info_available',
+                                is_rfi_info_available,
+                                'Whether or not RFI information is available')]
+
+            if is_rfi_info_available:
+                # Follow key/values only assigned if RFI info is avaiable
+                rfi_info_list = [
+                    Meta('rfi_mitigation_performed',
+                         burst.burst_rfi_info.rfi_mitigation_performed,
+                         ('Activation strategy of RFI mitigation'
+                          '["never", "BasedOnNoiseMeas", "always"]')),
+                    Meta('rfi_mitigation_domain',
+                         burst.burst_rfi_info.rfi_mitigation_domain,
+                         'Domain the RFI mitigation step was performed')
+                ]
+                rfi_qa_items_pol += rfi_info_list
+
+            # create HDF5 group for RFI info for current polarization
+            h5_rfi_path = f'{QA_PATH}/rfi_information/{burst.polarization}'
+            rfi_group = cslc_h5py_root.require_group(h5_rfi_path)
+
+            # write items to HDF5 and dict
+            _qa_items_to_h5_and_dict(rfi_group, self.rfi_dict, rfi_qa_items_pol)
+
+            # Take care of the burst RFI report information
+            if not is_rfi_info_available:
+                return
+
+            # Alias for readability
+            rfi_burst_report = burst.burst_rfi_info.rfi_burst_report
+
+            # Add the metadate of the burst RFI report
+            rfi_burst_report_list = [
+                Meta('swath',
+                     rfi_burst_report['swath'],
+                     'Swath of the burst'),
+                Meta('azimuth_time',
+                     datetime.datetime.strftime(rfi_burst_report['azimuthTime'],
+                                                TIME_STR_FMT),
+                    'Azimuth time of the burst report'),
+                Meta('in_band_out_band_power_ratio',
+                          rfi_burst_report['inBandOutBandPowerRatio'],
```

Comment:
BTW, for CSLC we decided to go the snake case (e.g. `in_band_out_band_power_ratio`) and not to use the camel case. Can you please modify this accordingly? Thanks

Reply:
The `rfi_burst_report` is from `Sentinel1BurstSlc.burst_rfi_info`. In s1reader side, I decided to use camel case (i.e. what S1 annotations follow) to simplify the XML parsing routine.